api/v1alpha1/serviceintentions_types.go

```diff
@@ -296,11 +334,64 @@ func (in IntentionPermissions) validate(path *field.Path) field.ErrorList {
 
 func (in *IntentionHTTPPermission) validate(path *field.Path) field.ErrorList {
 	var errs field.ErrorList
-	if invalidPathPrefix(in.PathPrefix) {
-		errs = append(errs, field.Invalid(path.Child("pathPrefix"), in.PathPrefix, `must begin with a '/'`))
+	pathParts := 0
+	if in.PathRegex != "" {
+		pathParts++
+	}
+	if in.PathPrefix != "" {
+		pathParts++
+		if invalidPathPrefix(in.PathPrefix) {
+			errs = append(errs, field.Invalid(path.Child("pathPrefix"), in.PathPrefix, `must begin with a '/'`))
+		}
+	}
+	if in.PathExact != "" {
+		pathParts++
+		if invalidPathPrefix(in.PathExact) {
+			errs = append(errs, field.Invalid(path.Child("pathExact"), in.PathExact, `must begin with a '/'`))
+		}
+	}
+	if pathParts > 1 {
+		asJSON, _ := json.Marshal(in)
+		errs = append(errs, field.Invalid(path, string(asJSON), `at most only one of pathExact, pathPrefix, or pathRegex may be configured.`))
+	}
+
+	found := make(map[string]struct{})
+	for i, method := range in.Methods {
+		methods := []string{
+			http.MethodGet,
+			http.MethodHead,
+			http.MethodPost,
+			http.MethodPut,
+			http.MethodPatch,
+			http.MethodDelete,
+			http.MethodConnect,
+			http.MethodOptions,
+			http.MethodTrace,
+		}
+		if !sliceContains(methods, method) {
+			errs = append(errs, field.Invalid(path.Child("methods").Index(i), method, notInSliceMessage(methods)))
+		}
+		if _, ok := found[method]; ok {
+			errs = append(errs, field.Invalid(path.Child("methods").Index(i), method, `method listed more than once.`))
+		}
+		found[method] = struct{}{}
 	}
-	if invalidPathPrefix(in.PathExact) {
-		errs = append(errs, field.Invalid(path.Child("pathExact"), in.PathExact, `must begin with a '/'`))
+	errs = append(errs, in.Header.validate(path.Child("header"))...)
+	return errs
+}
+
+func (in IntentionHTTPHeaderPermissions) validate(path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	for i, permission := range in {
+		hdrParts := 0
+		if permission.Present {
+			hdrParts++
+		}
+		hdrParts += numNotEmpty(permission.Exact, permission.Regex, permission.Prefix, permission.Suffix)
+		if hdrParts != 1 {
```

Comment:
`0` is valid here too.
```suggestion
		if hdrParts > 1 {
```

Reply:
good catch

Reply:
i kinda copied this from consul but i think they check this only if it exists so i miscopied it